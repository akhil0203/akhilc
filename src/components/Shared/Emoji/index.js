import React from 'react';
import PropTypes from 'prop-types';

const Emoji = props => {
  const { label, symbol } = props;

  return (
    <span
      className="emoji"
      role="img"
      aria-label={label}
      aria-hidden={label ? 'false' : 'true'}
    >
      {symbol}
    </span>
  );
};

Emoji.propTypes = {
  label: PropTypes.string,
  symbol: PropTypes.string.isRequired,
};

Emoji.defaultProps = {
  label: '',
};

export default Emoji;
