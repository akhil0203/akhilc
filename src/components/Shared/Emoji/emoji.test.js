import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Emoji from '.';

it('matches the snapshot', () => {
  const tree = renderer.create(<Emoji label="orangy" symbol="ðŸ¦§" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the Emoji component without crashing', () => {
  shallow(<Emoji label="orangy" symbol="ðŸ¦§" />);
});

it('sets the className', () => {
  const wrapper = shallow(<Emoji label="orangy" symbol="ðŸ¦§" />);

  expect(wrapper.prop('className')).toEqual('emoji');
});

it('sets the role', () => {
  const wrapper = shallow(<Emoji label="orangy" symbol="ðŸ¦§" />);

  expect(wrapper.prop('role')).toEqual('img');
});

it('sets the aria-label prop', () => {
  const wrapper = shallow(<Emoji label="orangy" symbol="ðŸ¦§" />);

  expect(wrapper.prop('aria-label')).toEqual('orangy');
});

it('sets the aria-hidden prop when supplied a label', () => {
  const wrapper = shallow(<Emoji label="orangy" symbol="ðŸ¦§" />);

  expect(wrapper.prop('aria-hidden')).toEqual('false');
});

it('sets the aria-hidden prop when supplied no label', () => {
  const wrapper = shallow(<Emoji symbol="ðŸ¦§" />);

  expect(wrapper.prop('aria-hidden')).toEqual('true');
});

it('sets the emoji', () => {
  const wrapper = shallow(<Emoji label="orangy" symbol="ðŸ¦§" />);

  expect(wrapper.prop('children')).toEqual('ðŸ¦§');
});

it('renders the html correctly', () => {
  const wrapper = shallow(<Emoji label="orangy" symbol="ðŸ¦§" />);

  expect(wrapper.html()).toEqual(
    '<span class="emoji" role="img" aria-label="orangy" aria-hidden="false">ðŸ¦§</span>',
  );
});
