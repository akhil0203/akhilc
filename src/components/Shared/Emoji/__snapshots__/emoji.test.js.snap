// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<span
  aria-hidden="false"
  aria-label="orangy"
  className="emoji"
  role="img"
>
  ðŸ¦§
</span>
`;
